Update README (#2)
google/project-euphonia-app: README.md

# Project Euphonia App

This repository provides open-source toolkits to build personalized speech recognition solutions, stemming from the broader [Project Euphonia](https://sites.research.google/euphonia/about/) initiative started by Google in 2019. This specific iteration focuses on enabling accurate speech-to-text transcription, particularly for individuals with conditions like Dysarthria.

> ## Intended Use

Author: I see value to first introduce the use to the repo before having him jumping on the setup. If you do not have strong opinion to move it after, I would keep it here.

> ## Intended Use

Reviewer: Good point! Let's keep it here 👍

cd project-euphonia-app
```

### Flutter-based mobile application

This component consists of a **Flutter-based mobile application**. The application consists on 2 main section:

- a section that allows users to record their voice saying specific phrases.
- a section to capture capture user speech and transcript into text relying on the trained model.

The application comes with a set of 100 default phrases located under `assets/phrases.txt`. You can customize or add more phrases. You can, for example, create training phrases for different languages.

The recorded speech data is stored in a **Firebase Storage** instance controlled by the developer [1].

#### Prerequisites

The application requires a [Firebase Storage](https://firebase.google.com/docs/storage). Please follow the following steps:

- Create a project from the [Firebase console](https://console.firebase.google.com/)
- [Create firebase storage](https://firebase.google.com/docs/storage/web/start) (Note: not "database", it needs to be "storage").
- Configure security rules as public. NOTE: This makes your files accessible by anyone. Consider adding autentication to secure your data.

> Install the following packages: [curl](https://curl.se/), [unzip](https://linux.die.net/man/1/unzip), [xz-utils](https://xz.tukaani.org/xz-utils/), [zip](https://linux.die.net/man/1/zip), [libglu1-mesa](https://packages.debian.org/buster/libglu1-mesa).

Reviewer: This and the other install steps can be avoided here by pointing users to [Flutter Setup page](https://docs.flutter.dev/get-started/install)

Author: I listed them here, because it was mentioned in the internal setup doc. Not sure if all those packages are mentioned in the Flutter Setup Page (e.g. libglu1-mesa). 

We can reduce the list to only libglu1-mesa.

WDYT?

Reviewer: It's a linux-specific setup step and mentioned in the [linux-setup](https://docs.flutter.dev/get-started/install/linux/android#development-tools) section

Install [Android Studio](https://developer.android.com/studio/install) 2023.3.1 (Jellyfish) or later to debug and compile Java or Kotlin code for Android. Flutter requires the full version of Android Studio.

Install the [Flutter SDK](https://docs.flutter.dev/get-started/install). When you run the current version of `flutter doctor`, it might list a different version of one of these packages. If it does, install the version it recommends.

### Installation

- Run `firebase login`
- Run `dart pub global activate flutterfire_cli`
- Configure Flutter project running `flutterfire configure --project=<project-id>`

For Android app use Android studio or run `flutter build apk` and `build/app/outputs/flutter-apk/app-release.apk` to install the app on the phone.
For iOS: Run `cd ios` and `run pod install`. Make sure mobile-provisioning profiles are present to install the app on-device.

### Train model

> This component consists in an Google Colab notebook to run the training. The Notebook will train ASR opensource model to racognize the speech of the user that recorded the training phrases.

Reviewer: ```suggestion
This component consists of a Google Colab notebook to run the training. The Notebook will train ASR an open source model to recognize the speech of the user that recorded the training phrases.
```

Open the notebook in the [training_colabs](https://github.com/google/project-euphonia-app/tree/main/training_colabs) and follow the steps.

### Web service that performs speech-to-text

> This component consist on a simple web application to serve the model trained in the previous step. The app is contenerized and can be run in Google Cloud Run.

Reviewer: ```suggestion
This component consist of a simple web application to serve the model trained in the previous step. The app is containerized and can be run in Google Cloud Run.
```

For Android app use Android studio or run `flutter build apk` and `build/app/outputs/flutter-apk/app-release.apk` to install the app on the phone.
For iOS: Run `cd ios` and `run pod install`. Make sure mobile-provisioning profiles are present to install the app on-device.

### Train model

This component consists in an Google Colab notebook to run the training. The Notebook will train ASR opensource model to racognize the speech of the user that recorded the training phrases.

Open the notebook in the [training_colabs](https://github.com/google/project-euphonia-app/tree/main/training_colabs) and follow the steps.

### Web service that performs speech-to-text

This component consist on a simple web application to serve the model trained in the previous step. The app is contenerized and can be run in Google Cloud Run.

#### Deploy

Copy the trained model (e.g. `pytorch_model.bin`) into the `api/custom_tiny_whisper_model/` folder. Place custom tiny whisper model in the custom_tiny_whisper_model directory as pytorch_model.bin.

Deploy the web app into Google Cloud Run following steps reported in the [app README](api/).

## Usege

> Following all [Setup](#setup) steps you installed the Euphonia App on your smarphone and the api on Goocle Cloud Run. You can now set the URL of the Google Cloud Run instance in the app configuration and start transcribing your speech.

Reviewer: ```suggestion
Following all [Setup](#setup) steps you installed the Project Euphonia App on your smartphone and the api on Google Cloud Run. You can now set the URL of the Google Cloud Run instance in the app Settings and start transcribing your speech.
```
